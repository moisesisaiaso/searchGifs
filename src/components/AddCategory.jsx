import { useState } from "react";

export const AddCategory = ({ onAddCategory }) => {
    const [inputValue, setInputValue] = useState(""); //* las funciones de estados contienen un callback que se puede resumir de manera implicita cuando su proceso es simple pero en sí el collback de esta función se argumenta con el estado es decir tiene como propiedad el estado

    const onInputChange = ({ target }) => {
        setInputValue(target.value);
    };

    const onSubmit = (event) => {
        event.preventDefault(); // esto evita el refresh del navegador cuando se envia el formulario
        if (inputValue.trim().length <= 1) return;

        onAddCategory(inputValue.trim());
        setInputValue("");
    };
    return (
        <form onSubmit={onSubmit}>
            <input
                type="text"
                placeholder="Buscar Gifs"
                value={
                    inputValue
                } /* esto me asigna un valor estatico al campo que no podrá cambiar al querer ingresar un nuevo valor */
                onChange={
                    onInputChange
                } /* con el onChange le estamos diciendo al atributo value que va a ser dinamico, de esta forma atraves de la propiedad target.value le damos un valor nuevo que se actualiza cada vez que se renderiza el componente y como el estado se actualiza en cada teclada pues cada caracter se mostrará de manera dinamica  ya que cada vez que el estado se actualiza se renderiza el componente*/
            />
        </form>
    );
};
