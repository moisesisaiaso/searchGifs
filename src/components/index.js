export * from "./AddCategory"; // exporta todo lo que haya en /AddCategory
export * from "./GifGrid";
export * from "./GifItem";

//* esto se le suelve llamar archivo de barril y me permite agrupar las importaciones de por ejemplo los componentes, igual que se importa un useState y un useEffect desde una misma importación y ruta

//^ de esta manera agrupo todo al index y cuando importe los componentes lo puedo hacer directamente desde la ruta /components sin necesidad de hacer referencia a el archivo index ya que es redundante y lo señala de manera implicita

//? es recomendable hacer esto cuando hay varios archivos en una carpeta donde se están exportando funciones o componentes
