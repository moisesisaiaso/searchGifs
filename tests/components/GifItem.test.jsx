import { GifItem } from "../../src/components/GifItem";

import { render } from "@testing-library/react";

describe("pruebas en <GifItem />", () => {
    const title = "Batman";
    const url = "https://batmanGifPrueba.gif";

    test("hacer match en el snapshot", () => {
        const { container } = render(<GifItem />);

        expect(container).toMatchSnapshot();
    });
});
