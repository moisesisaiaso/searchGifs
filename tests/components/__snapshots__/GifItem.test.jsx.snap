// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <GifItem /> hacer match en el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Batman"
      src="https://batmanGifPrueba.gif"
    />
    <p>
      Batman
    </p>
  </div>
</div>
`;
